code/gui/griddisplay.cpp

```diff
@@ -135,6 +195,13 @@ void GridDisplay::clear_selection() {
 }
 
 void GridDisplay::show_grid() {
+    m_row_count = (m_image_viewer->height() / 2) / GRID_SIZE;
+    m_column_count = (m_image_viewer->width() / 2) / GRID_SIZE;
+    m_scene->setSceneRect(QRect(0, 0, m_column_count * GRID_SIZE, m_row_count * GRID_SIZE));
+    m_view->resize(m_column_count * GRID_SIZE, m_row_count * GRID_SIZE);
+    m_view->move(m_image_viewer->width() / 2 - m_scene->width() / 2, m_image_viewer->height() / 2 - m_scene->height() / 2);
+    m_button = array2d<GridButton *>(m_column_count, m_row_count);
```

Comment:
these arrays should be initialized once in the constructor only

Reply:
I wasn't able to make that work since we don't know the size of `m_image_viewer` yet so `m_row_count` and `m_column_count` are still 0 in the constructor.